### PR TITLE
Support writing strings to stream

### PR DESCRIPTION
**Pull request by kasperisager**
Currently fails on the native side:

```
node[1859]: ../src/node_buffer.cc:246:char* node::Buffer::Data(v8::Local<v8::Value>): Assertion `val->IsArrayBufferView()' failed.
[157](https://github.com/mafintosh/libudx/runs/6270814260?check_suite_focus=true#step:5:157)
 1: 0xb09980 node::Abort() [node]
[158](https://github.com/mafintosh/libudx/runs/6270814260?check_suite_focus=true#step:5:158)
 2: 0xb099fe  [node]
[159](https://github.com/mafintosh/libudx/runs/6270814260?check_suite_focus=true#step:5:159)
 3: 0xae06ca node::Buffer::Data(v8::Local<v8::Value>) [node]
[160](https://github.com/mafintosh/libudx/runs/6270814260?check_suite_focus=true#step:5:160)
 4: 0xad271e napi_get_buffer_info [node]
[161](https://github.com/mafintosh/libudx/runs/6270814260?check_suite_focus=true#step:5:161)
 5: 0x7fe6f8c95968 udx_napi_stream_write [/home/runner/work/libudx/libudx/build/Release/udx.node]
[162](https://github.com/mafintosh/libudx/runs/6270814260?check_suite_focus=true#step:5:162)
 6: 0xab44dd  [node]
[163](https://github.com/mafintosh/libudx/runs/6270814260?check_suite_focus=true#step:5:163)
 7: 0xd53d8e  [node]
[164](https://github.com/mafintosh/libudx/runs/6270814260?check_suite_focus=true#step:5:164)
 8: 0xd551af v8::internal::Builtin_HandleApiCall(int, unsigned long*, v8::internal::Isolate*) [node]
[165](https://github.com/mafintosh/libudx/runs/6270814260?check_suite_focus=true#step:5:165)
 9: 0x15f0bf9  [node]
```

### CODE REVIEW

**Comment by mafintosh** [comment]
Good catch, we can use the mapWritable option in streamx to fix that easily

File: lib/stream.js [hunk 1]
```diff
@@ -255,3 +256,7 @@ module.exports = class UDXStream extends streamx.Duplex {
 }
 
 function noop () {}
+
+function toBuffer (data) {
+  return b4a.isBuffer(data) ? data : b4a.from(data)
```
Comment (mafintosh) [inline lib/stream.js:261]: i think typeof data === string ? b4a.from(data) : data is a bit safer as it's more conservative (would prefer an array crashing the process with an assertion later on)

Comment (mafintosh) [reply lib/stream.js:261]: can you b4a the other allocs in here, might as well

File: test/stream.js [hunk 2]
```diff
@@ -359,9 +359,32 @@ test('close socket on stream close', async function (t) {
...
+    .on('close', function () {
+      t.pass('b closed')
+    })
+    .end('hello world')
+})
+
+test('write object', async function (t) {
+  const [a, b] = makeTwoStreams(t)
+
+  t.exception.all(() => a.write({ hello: 'world' }))
```
Comment (mafintosh) [inline test/stream.js:386]: imo we don't need to test this, the user should not do this.

Comment (mafintosh) [reply test/stream.js:386]: as in, if write throws, all bets are off (undefined behaivour)

Comment (kasperisager) [reply test/stream.js]: The issue, I guess, is that it doesn't throw; the process crashes with a native assertion error.

Comment (mafintosh) [reply test/stream.js]: that's fine also honestly - ie don't do that

**Pull request merged**